### PR TITLE
Docs: publish docs for SelectPanel v2 and Dialog v2

### PR DESCRIPTION
**Pull request by siddharthkp**
Based on Tooltip v2, added `docsId` for Dialog v2 and SelectPanel v2

### CODE REVIEW

File: packages/react/src/drafts/SelectPanel2/SelectPanel.docs.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
-  "id": "selectpanel_v2",
-  "name": "SelectPanel",
+  "id": "select_panel_v2",
+  "docsId": "select_panel",
+  "name": "SelectPanel v2",
```
Comment (siddharthkp) [inline packages/react/src/drafts/SelectPanel2/SelectPanel.docs.json:4]: @broccolini Should we call this `SelectPanel v2` or just `SelectPanel`?

Seems like [Tooltip v2 is titled just Tooltip in the docs](https://primer.style/components/tooltip/react/beta), even though [docs.json says "Tooltipv2"](https://github.com/primer/react/blob/main/packages/react/src/TooltipV2/Tooltip.docs.json#L3)

File: packages/react/src/SelectPanel/SelectPanel.docs.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "id": "select_panel",
+  "docsId": "select_panel",
```
Comment (siddharthkp) [inline packages/react/src/SelectPanel/SelectPanel.docs.json:3]: @broccolinisoup Do we need docsId on both v1 and v2 docs.json?

File: packages/react/src/drafts/SelectPanel2/SelectPanel.docs.json (hunk 1 above)
Comment (broccolinisoup) [reply packages/react/src/drafts/SelectPanel2/SelectPanel.docs.json:4]: Great catch! - And this is a [temporary hack](https://github.com/primer/design/pull/752/commits/116966476a1ff08de10aee82d758ca95014fad96) until we release https://github.com/primer/react/pull/4483 

So, we should keep the name as `SelectPanel` 😊

File: packages/react/src/SelectPanel/SelectPanel.docs.json (hunk 2 above)
Comment (broccolinisoup) [reply packages/react/src/SelectPanel/SelectPanel.docs.json:3]: Yes please. We need the unique docsId in all components that we want to group together.

File: packages/react/src/Dialog/Dialog.docs.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
-  "id": "drafts_dialog",
-  "name": "Dialog",
+  "id": "dialog_v2",
+  "docsId": "dialog",
+  "name": "Dialog v2",
```
Comment (broccolinisoup) [inline packages/react/src/Dialog/Dialog.docs.json:4]: We can keep the name as `Dialog` here as well. 

The challenge with Dialog is that the story ids. Currently https://primer.style/components/dialog/react/alpha show the alpha dialog but the examples are belong to the draft dialog 😵  (If I am not mistaken) 

Can we take the dialog v2 stories under draft? and remove that `v1` from the alpha dialog to correct the story ids? I also remember there was a confusion around this and this is why we made this change, right? So we might maybe need to come up with custom solution for dialog if we don't want to revert the stories back.

**Review by broccolinisoup — COMMENTED** [review]
Looks good! Left some comments 🙌🏻

**Comment by siddharthkp** [comment]
@broccolinisoup Updated the names!

**Review by broccolinisoup — APPROVED** [review]

**Pull request merged**